pkg/didcomm/messenger/messenger.go

```diff
@@ -77,16 +81,84 @@ func (m *Messenger) HandleInbound(msg service.DIDCommMsgMap, myDID, theirDID str
 		return fmt.Errorf("threadID: %w", err)
 	}
 
+	if err := m.populateMetadata(thID, msg); err != nil {
+		return fmt.Errorf("with metadata: %w", err)
+	}
+
 	// saves message payload
 	return m.saveRecord(msg.ID(), record{MyDID: myDID, TheirDID: theirDID, ThreadID: thID})
 }
 
+func (m *Messenger) saveMetadata(msg service.DIDCommMsgMap) error {
+	if msg[jsonThread] == nil {
```

Comment:
I would assume a protocol may add metadata to a message when initiating a protocol. The `~thread` decorator is not normally inserted in those cases. This suggests to me that the metadata should be contained outside the `~thread`.